### ISSUE TITLE
"No SRV Records" for many servers

### ISSUE THREAD

**Issue by hex-m** [matrix-org/matrix-federation-tester#104]
I noticed that the tester did not seem to recognize my SRV records. Comparing with other servers it seems that the tester may have a bug? Big servers like [matrix.org](https://federationtester.matrix.org/#matrix.org) and [tchncs.de](https://federationtester.matrix.org/#tchncs.de) are shown to have no SRV records.

But they actually do:
```
$ dig +short SRV _matrix._tcp.matrix.org
10 5 8443 matrix-federation.matrix.org.cdn.cloudflare.net.
```

Interestingly, the SRV-record of [chat.weho.st](https://federationtester.matrix.org/#chat.weho.st) is recognized,  although it looks exactly like the prior examples to me.

```
$ dig +short SRV _matrix._tcp.chat.weho.st
0 10 8448 chat.weho.st.
```

**Comment by aaronraimist** [comment]
It is working properly. Read https://matrix.org/docs/spec/server_server/r0.1.4#resolving-server-names.

**Comment by erikjohnston** [comment]
To be more explicit: matrix.org uses a `.well-known`, which is looked up before checking the SRV records. A lot of servers will have been using SRV and then migrated to `.well-known` and have just left their SRV records up.

It is a bit confusing, and maybe the federation tester UI should have a little note about that, or display what SRV records were checked (if any).

**Comment by erikjohnston** [comment]
We do a SRV lookup on the record returned by `.well-known` (unless it contains an explicit port or IP), its just that most people don't use both a `.well-known` and an SRV record :slightly_smiling_face:

**Comment by daenney** [comment]
I think this is a really confusing UX thuogh.

"No SRV records" to me means "we checked for the _matrix._tcp.<delegated_hostname> record and didn't find any". Not so much "oh we found a well-known so we skipped it entirely".

**Comment by aaronraimist** [comment]
@daenney if you read the message just above, you’ll see it is not skipped entirely, it just doesn’t work how you think it does.

**Comment by daenney** [comment]
That's fair. But then the frontend is still a bit broken from my point of view. "No SRV records"  should mean "we did the equivalent of dig -t SRV" and got a NODATA/NXDOMAIN.

I think what the frontend is trying to convey is "we didn't check for SRV records in this case, because we found a well-known that contained a host:port tuple". Which is perfectly reasonable, since that's how federation is specified to work.
